graph-commons/src/main/scala/io/renku/graph/model/events.scala

```diff
@@ -220,6 +247,65 @@ object events {
     }
   }
 
+  final case class EventInfo(eventId:         EventId,
+                             projectPath:     projects.Path,
+                             status:          EventStatus,
+                             eventDate:       EventDate,
+                             executionDate:   ExecutionDate,
+                             maybeMessage:    Option[EventMessage],
+                             processingTimes: List[StatusProcessingTime]
+  )
+
+  final case class StatusProcessingTime(status: EventStatus, processingTime: EventProcessingTime)
+  object StatusProcessingTime {
+    implicit val jsonEncoder: Encoder[StatusProcessingTime] = {
+      import io.circe.literal._
+
+      processingTime => json"""{
+        "status":         ${processingTime.status},
+        "processingTime": ${processingTime.processingTime}
+      }"""
+    }
+
+    implicit val jsonDecoder: Decoder[StatusProcessingTime] =
+      Decoder.instance { cursor =>
+        for {
+          status <- cursor.downField("status").as[EventStatus]
+          time   <- cursor.downField("processingTime").as[EventProcessingTime]
+        } yield StatusProcessingTime(status, time)
+      }
+  }
+
+  object EventInfo {
+
+    implicit val jsonDecoder: Decoder[EventInfo] =
+      Decoder.instance { cursor =>
+        for {
+          id              <- cursor.downField("id").as[EventId]
+          path            <- cursor.downField("projectPath").as[projects.Path]
+          status          <- cursor.downField("status").as[EventStatus]
+          processingTimes <- cursor.downField("processingTimes").as[List[StatusProcessingTime]]
+          date            <- cursor.downField("date").as[EventDate]
+          executionDate   <- cursor.downField("executionDate").as[ExecutionDate]
+        } yield EventInfo(id, path, status, date, executionDate, None, processingTimes)
+      }
+
+    implicit val jsonEncoder: Encoder[EventInfo] = eventInfo => {
```

Comment:
I'm also fine to move them. For me it is ok to put them next to each other. The reason is that this case class defines what data can be consumed from the endpoint and here there is a "canonical" json format for it. So I would put the codec (decoder and encoder) close to each other, because they are supposed to complement each other. But not a strong opinion. Where would you prefer the encoder to be defined? Maybe as a private value in the `EventsEndpoint` companion?

Reply:
Yes, for me the endpoint seems to be probably a better place.